dbld/images/helpers/dependencies.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+set -e
+set -x
+
+function install_cmake() {
+    CMAKE_VERSION=3.12.2
+    CMAKE_SHORT_VERSION=$(echo ${CMAKE_VERSION} | cut -d"." -f1-2)
+    download_target "https://cmake.org/files/v${CMAKE_SHORT_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.sh" /tmp/cmake.sh
+    chmod +x /tmp/cmake.sh
+    mkdir -p /opt/cmake
+    /tmp/cmake.sh --skip-license --prefix=/opt/cmake/
+    ln -s /opt/cmake/bin/cmake /usr/bin/cmake
+    rm -rf /tmp/cmake.sh
+}
+
+function install_criterion() {
+    CRITERION_VERSION=2.3.2
+    download_target "https://github.com/Snaipe/Criterion/releases/download/v${CRITERION_VERSION}/criterion-v${CRITERION_VERSION}-linux-x86_64.tar.bz2" /tmp/criterion.tar.bz2
+    tar xvjf /tmp/criterion.tar.bz2 --strip 1 -C /usr
+    rm -rf /tmp/criterion.tar.bz2
+}
+
+function install_gosu() {
+    GOSU_VERSION=1.10
+    ARCHITECTURE=$1
+    cat /helpers/gosu.pubkey | gpg --import
+    download_target "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-${ARCHITECTURE}" /usr/local/bin/gosu
+    download_target "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-${ARCHITECTURE}.asc" /usr/local/bin/gosu.asc
+    gpg --verify /usr/local/bin/gosu.asc
+    rm /usr/local/bin/gosu.asc
+    chmod +x /usr/local/bin/gosu
+}
+
+function install_gradle {
+    GRADLE_VERSION=4.10
+    download_target "https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip" /tmp/gradle.zip
+    mkdir -p /opt/gradle
+    unzip -d /opt/gradle /tmp/gradle.zip
+    rm -rf /tmp/gradle.zip
+    ln -s /opt/gradle/gradle-${GRADLE_VERSION}/bin/gradle /usr/bin/gradle
+}
+
+function install_libmaxminddb() {
+    LIBMAXMINDDB_VERSION=1.3.2
+    download_target "https://github.com/maxmind/libmaxminddb/releases/download/${LIBMAXMINDDB_VERSION}/libmaxminddb-${LIBMAXMINDDB_VERSION}.tar.gz" /tmp/libmaxminddb.tar.gz
+    cd /tmp/
+    tar xvf /tmp/libmaxminddb.tar.gz
+    cd /tmp/libmaxminddb-${LIBMAXMINDDB_VERSION}
+    ./configure
+    make
+    make install
+    ldconfig
+    rm -rf /tmp/libmaxminddb.tar.gz /tmp/libmaxminddb-${LIBMAXMINDDB_VERSION}
+}
```

Comment:
runtime dependencies should not be installed from source, unless strictly necessary. libmaxminddb is available in ubuntu repositories, so we should just use that. Ah.. I see that you were only doing this on platforms where this is not available.

For non-critical stuff, I'd simply not compile them on the platform, for critical stuff (I am not sure there's any), this could be an option, but then we would probably have to supply the binaries bundled.

For dev tools, it is ok to pull in 3rd party tools, but again I'd try to avoid that if possible.

Reply:
We agreed on this and the latest version dropped the private compilations during build, so this is resolved.